src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -0,0 +1,112 @@
+import { Buffer } from "buffer";
+import {
+  Paginate,
+  Cip30DataSignature,
+  IWalletInfo,
+} from "@fabianbormann/cardano-peer-connect/dist/src/types";
+import { CardanoPeerConnect } from "@fabianbormann/cardano-peer-connect";
+import { AriesAgent } from "../../agent/agent";
+import { IdentifierType } from "../../agent/services/identifierService.types";
+
+class IdentityWalletConnect extends CardanoPeerConnect {
+  getIdentifierId: () => Promise<string>;
+  signDataWithIdentifier: (
+    identifierId: string,
+    payload: string
+  ) => Promise<string>;
+  generateOobi: (identifierId: string) => Promise<string>;
+
+  constructor(
+    walletInfo: IWalletInfo,
+    seed: string | null,
+    announce: string[],
+    discoverySeed?: string | null
+  ) {
+    super(walletInfo, {
+      seed: seed,
+      announce: announce,
+      discoverySeed: discoverySeed,
+      logLevel: "info",
+    });
+
+    this.getIdentifierId = async (): Promise<string> => {
+      const identifiers = await AriesAgent.agent.identifiers.getIdentifiers();
+      if (identifiers && identifiers.length > 0) {
+        for (const identifier of identifiers) {
+          if (identifier.method === IdentifierType.KERI) {
+            return identifier.id;
+          }
+        }
+        throw new Error("No KERI identifier stored");
+      } else {
+        throw new Error("No identifier stored");
+      }
+    };
+
+    this.signDataWithIdentifier = async (
+      identifierId: string,
+      payload: string
+    ): Promise<string> => {
+      const signer = await AriesAgent.agent.identifiers.getSigner(identifierId);
```

Comment:
I think we can keep a state variable to cache the mapping between the `identifierId` and `signer` so we don't pull it continuously.